### ISSUE TITLE
LSP servers not starting

### ISSUE THREAD

**Issue by bhankas** [doomemacs/doomemacs#6771]
### What did you expect to happen?

When launching files with major mode/modules supporting `+lsp` flag, language server is automatically started by eglot and connected to.

This was tested for `nix` as well as `elixir` files. Until update on 10th September, this was confirmed working with `Eglot`, and no configuration beyond enabling the module and adding `+lsp` flag.

### What actually happened?

SInce 10th Sept update for doom-emacs (continued to latest as of this issue), such modes fail to start LSP with below error:

```
Error (nix-mode-local-vars-mode): Error running hook "lsp!" because: (void-function 'lsp-deferred)
```

### Describe your attempts to resolve the issue

Reverted to 0172b180a901cb515e8a6d2d0272535a870853dd and worked as before.

### Steps to reproduce

1. Update to latest doom-emacs
2. Enable Nix module with `+lsp` flag
3. Install rnix-lsp
4. `doom-sync`
5. Open any .nix file

* I understand I am on Emacs 29, but I confirmed same behavior with a friend on Emacs 28.

### System Information

https://pastebin.com/4AfiRBKs

**Comment by matasar** [comment]
I think this is because `modulep!` isn't behaving as expected. On my machine it calls `(doom-module-from-path)` which doesn't resolve to the right module.

**Comment by bhankas** [comment]
45a66cda606085f67db977034d9f3d666bdd1934 this is the only relavant commit in recent history I could find. Can we test by reverting it alone? Or will it cause some cascade elsewhere?

**Comment by bhankas** [comment]
Updated with temporary fix:
```
cd ~/.emacs.d
git reset --hard 0172b180a901cb515e8a6d2d0272535a870853dd
doom sync
```

**Comment by rymndhng** [comment]
I'm running into this issue on Emacs 28 as well.

I have `+eglot` enabled and it appears that this expression `(modulep! +eglot)`  is evaluating to false when it should be true. See line: https://github.com/doomemacs/doomemacs/blob/29e30e8af4b937cd0d4c9b5eb900205f22bddf7b/modules/tools/lsp/autoload/common.el#L7

**Comment by mood418** [comment]
> I'm running into this issue on Emacs 28 as well.
> 
> I have `+eglot` enabled and it appears that this expression `(modulep! +eglot)` is evaluating to false when it should be true. See line:
> 
> https://github.com/doomemacs/doomemacs/blob/29e30e8af4b937cd0d4c9b5eb900205f22bddf7b/modules/tools/lsp/autoload/common.el#L7

Change this line  to `(if (modulep! :tools lsp +eglot)' seems solve this issue.

**Comment by elken** [comment]
Seems like there's an underlying issue between `modulep!` and autoloads. Have raised a PR to try and resolve this one at least though.

**Comment by bhankas** [comment]
As another QnD hack, I just added this snippet to my config and its working :p

```
(defun lsp-deferred ()
    (eglot-ensure)
```

**Comment by hlissner** [comment]
As of 7161f27bb3283761c7a7e653de187e2eb8e3d191, this should be fixed. Let me know if that isn't the case and I'll reopen this. Sorry for the trouble!

**Comment by rymndhng** [comment]
@hlissner the fix worked 11 days ago --  but I noticed that with the latest version of doom (95bdc3b56), the issue has returned.

**Comment by rymndhng** [comment]
^ and it looks like it's fixed again as of `285b460c8`
